### PR TITLE
improve search function, search results that are local to the devhub …

### PR DESCRIPTION
**Pull request by w8896699**
Local resources will show up while other resources still loading.

### CODE REVIEW

File: app-web/src/pages/index.js [hunk 1]
```diff
@@ -407,6 +413,7 @@ export const Index = ({
         resultCount={totalSearchResults}
       />
       <Main>
+        {windowHasQuery && siphonResources}
```
Comment (w8896699) [inline app-web/src/pages/index.js]: if search without query, the location.search will be ?q='' which will lead a bug. (cuz windowHasQuery is True)

File: app-web/src/components/Home/ResourcePreview.js [hunk 2]
```diff
@@ -92,6 +92,14 @@ export const ResourcePreview = ({ title, link, resources, filters, amountToShow,
   let [activeFilters, setActiveFilters] = useState(['All']);
   const extraItemsToShow = 6; //two more row of card in the page after click 'see more'
 
+  useEffect(() => {
+    setResources(resources);
+
+    return () => {
+      setResources();
```
Comment (patricksimonian) [inline app-web/src/components/Home/ResourcePreview.js]: setResources to `undefined`?

File: app-web/src/pages/index.js [hunk 3]
```diff
@@ -251,7 +251,11 @@ export const Index = ({
     DYNAMIC_TOPIC_PATHS.featured,
     FEATURED_CONTENT,
   );
-
+  // dynamic sources all load at different times, this function returns false when all have completed loading
+  let [searchSourcesLoading, setLoading] = useState(searchGate.loading);
+  if (!!searchSourceResults.rocketchat) {
```
Comment (patricksimonian) [inline app-web/src/pages/index.js]: what does this step do?

File: app-web/src/pages/index.js [hunk 4]
```diff
@@ -407,7 +413,12 @@ export const Index = ({
         resultCount={totalSearchResults}
       />
       <Main>
-        {windowHasQuery && searchSourcesLoading ? <Loading message="loading" /> : content}
+        {windowHasQuery && queryParam.q !== '' && siphonResources}
```
Comment (patricksimonian) [inline app-web/src/pages/index.js]: can you double check if we are already making this check in `queryisEmpty`

**Pull request merged**